### DISCUSSION TITLE
Typescript -  Assertion and Conditions types?

### DISCUSSION THREAD

**Discussion by onserv** [bergie/passport-saml#747] [Ideas]
Help request

After upgrading to current passport-saml and @types/passport-saml my deriving expires in code broke..  Likely missing something basic but seems to me that the type package is not complete for Assertion and Conditions...?
...
return new SamlStrategy.Strategy(samlConfig, (profile: SamlStrategy.Profile, done: VerifiedCallback) => {
		const notBefore = new Date(profile.getAssertion().Assertion.Conditions[0].$.NotBefore)
		const notOnOrAfter = new Date(samlResponse.getAssertion().Assertion.Conditions[0].$.NotOnOrAfter)
		const expiresIn = (+notOnOrAfter - +notBefore) / 1000
		return done(null, { iagProfile: convertSamlResponseToIagProfile(profile), expiresIn }, null)
	})
...
Are we missing Type declarations ?
	
src/strategies/passport-saml-strategy.ts:113:63 - error TS2339: Property 'Conditions' does not exist on type 'unknown'.
113     const notBefore = new Date(profile.getAssertion().Assertion.Conditions[0].$.NotBefore)
                                                                    ~~~~~~~~~~
src/strategies/passport-saml-strategy.ts:114:66 - error TS2339: Property 'Conditions' does not exist on type 'unknown'.
114     const notOnOrAfter = new Date(profile.getAssertion().Assertion.Conditions[0].$.NotOnOrAfter)

Suggestions appreciated.
                                                                       ~~~~~~~~~~

**Comment by onserv** [comment]
I resolved like this...
return new SamlStrategy.Strategy(samlConfig, (profile: SamlStrategy.Profile, done: VerifiedCallback) => {
		// TODO when passport-saml types are enhanced we should not have to cast
		const tmpAssertions: any = profile.getAssertion().Assertion
		const notBefore = new Date(tmpAssertions.Conditions[0].$.NotBefore)
		const notOnOrAfter = new Date(tmpAssertions.Conditions[0].$.NotOnOrAfter)
		const expiresIn = (+notOnOrAfter - +notBefore) / 1000

		return done(null, { iagProfile: convertSamlResponseToIagProfile(profile), expiresIn }, null)
	})

**Comment by cjbarth** [comment]
The latest version of `passport-saml` doesn't need external types. You shouldn't have the `@types/passport-saml` package installed.
